Aptoide/aptoide-client-v8
# [ASV-1820] Verify your wallet

**What does this PR do?**

   This PR aims at implementing the verify your wallet flow in the promotions claim flow. With this new flow, we removed the captcha.
Also took the chance to refactor some code that was related with the promotions claim flow.

**Database changed?**

   No

**Where should the reviewer start?**

- [ ] ClaimPromotionsPresenter.java

**How should this be manually tested?**

  Since this was not implemented on the wallet by the appcoins team, they gave us a wallet version that would handle our intents, with 3 buttons to simulate their flow:
 - cancel - user canceled, so we should show an error message.
 - ok - it was verified, so we should automatically claim the app again.
 - fail - any error that was not handled by the wallet.

Also, since we have a bug on AppView where you can not claim the promotion related with migration (in case the wallet is an update), I recommend you to install Jane casino migrated version and use the rewrites i added in the ticket.

Documentation here: https://aptoide.atlassian.net/wiki/spaces/AN/pages/1340768263/Verify+your+wallet

**What are the relevant tickets?**

  Tickets related to this pull-request: [ASV-1820](https://aptoide.atlassian.net/browse/ASV-1820)

**Questions:**

   Does this add new dependencies which need to be added? (Eg. new keys, etc.) 



**Code Review Checklist**
